### PR TITLE
Add X-Powered-By header.

### PR DESCRIPTION
**Pull request by arunoda**
See:

```
sh-3.2$ curl -v http://localhost:3000
* Rebuilt URL to: http://localhost:3000/
*   Trying ::1...
* Connected to localhost (::1) port 3000 (#0)
> GET / HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/7.49.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/html
< Content-Length: 3172
-------------------------------
< X-Powered-By: Next.js 1.2.3
-------------------------------
< Date: Sat, 17 Dec 2016 11:38:52 GMT
< Connection: keep-alive
```

### CODE REVIEW

**Comment by STRML** [comment]
This feels like a bad idea - in general you don't want to leak information about the server in any production app - the framework name is bad enough, the exact version is worse because it makes it that much easier to target exploits.

File: server/render.js [hunk 1]
```diff
@@ -102,6 +104,7 @@ export async function renderErrorJSON (err, res, { dir = process.cwd(), dev = fa
 export function sendHTML (res, html) {
   res.setHeader('Content-Type', 'text/html')
   res.setHeader('Content-Length', Buffer.byteLength(html))
+  res.setHeader('X-Powered-By', `Next.js ${pkg.version}`)
```
Comment (sedubois) [inline server/render.js]: could use `pkg.name` to DRY

**Comment by jamo** [comment]
Yeah, +1 for obfuscating the version. Tho if the page downloads `next.js/<version>/next.min.js` it's pretty obvious which version and framework is being used...

**Comment by arunoda** [comment]
Securing apps via `obfuscating` is a good start but it doesn't allow hold any good hacker to do his/her thing. (If there's an issue).

This is something Express does [all the times](https://github.com/expressjs/express/search?utf8=%E2%9C%93&q=X-Powered-By).

What they have a way to turn it off. I'll think about something like that.

**Comment by rauchg** [comment]
All major web servers do this. 
It allows us to in the future actually _responsibly disclose_ security problems!
That said, @arunoda it'd be helpful to introduce a config flag `poweredByHeader: false` to turn it off

**Comment by arunoda** [comment]
@rauchg That's a good suggestion.
I'll hold this PR and do it once we've land #222

**Comment by arunoda** [comment]
@rauchg check now.
We could disable setting the header.
Additionally, we could access the config via `app.config` if the user uses #310

**Comment by rauchg** [comment]
And a test would be nice to add as well!

**Comment by arunoda** [comment]
@rauchg added some test cases.

**Pull request merged**